Automattic/jetpack
# tools: Add config for the --type for tools adding change entries

#### Changes proposed in this Pull Request:

In a few places we have it special cased to do `--type=other` instead of
`--type=changed` when auto-adding a change entry for plugins/jetpack.

Instead, let's make that a configuration in composer.json so if we have
other plugins that copy Jetpack in the future then we don't have to find
all the scripts with special casing.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Edit `projects/packages/sync/composer.json` to change `.extra.branch-alias.dev-master`.
* Run `tools/check-intra-monorepo-deps.sh -uv`.
* See that it generates correct change files, both for Jetpack and for other plugins.
